src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/HotReload/ProjectHotReloadSession.cs

```diff
@@ -155,25 +230,30 @@ public async ValueTask ApplyUpdatesAsync(ImmutableArray<ManagedHotReloadUpdate>
                 }
             }
         }
-        catch (Exception ex)
+        catch (Exception e) when (LogAndPropagate(e, cancellationToken))
+        {
+            // unreachable
+        }
+    }
+
+    private bool LogAndPropagate(Exception e, CancellationToken cancellationToken)
+    {
+        if (e is not OperationCanceledException)
         {
             WriteToOutputWindow(
-                string.Format(Resources.HotReloadApplyUpdatesFailure, $"{ex.GetType()}: {ex.Message}"),
+                string.Format(Resources.HotReloadApplyUpdatesFailure, $"{e.GetType()}: {e.Message}"),
```

Comment:
Are we sure this won't lose information from `InnerException` or `InnerExceptions`? Won't be very helpful if this is `TargetInvocationException` or `AggregateException` for example.

Reply:
I just removed the rethrow here. I'd leave improvements to exception handling to another PR. 
Ideally, this would also report non-fatal Watson.